src/AutoRest.CSharp/Common/Output/Models/Types/GenericParameterTypeProvider.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright(c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT License.
+
+using AutoRest.CSharp.Common.Input;
+using AutoRest.CSharp.Input.Source;
+
+namespace AutoRest.CSharp.Output.Models.Types
+{
+    internal class GenericParameterTypeProvider : TypeProvider
```

Comment:
I think we should make this sealed since this has its own special logic in `CodeWriter`

Reply:
Or we could always just get a generic parameter definition from the compiler so that we do not have to have this class.